### PR TITLE
Change test name so it triggers in CI

### PR DESCRIPTION
**Pull request by adarshsanjeev**
As the name of the class did not end or start with "Test", `CalciteSelectQueryMSQTest` was not triggered in CI. This PR renames the test.

<hr>

This PR has:

- [x] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [ ] added documentation for new or modified features or behaviors.
- [ ] a release note entry in the PR description.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [ ] been tested in a test Druid cluster.

### CODE REVIEW

File: extensions-core/multi-stage-query/src/test/java/org/apache/druid/msq/test/CalciteSelectQueryMSQTest.java [hunk 1]
```diff
@@ -161,4 +165,19 @@ public void testQueryWithMoreThanMaxNumericInFilter()
 
   }
 
+  @Test
+  @Override
+  public void testArrayAggQueryOnComplexDatatypes()
```
Comment (paul-rogers) [inline extensions-core/multi-stage-query/src/test/java/org/apache/druid/msq/test/CalciteSelectQueryMSQTest.java:170]: Would be worth including an explanation that this feature does not work on MSQ, but does work with "classic" queries. Might be wroth also adding links between the tests to cross-reference the MSQ vs. "classic" case.

That way, if this ever changes in the future, folks can sort out where the two copies of the test live.

**Review by paul-rogers — COMMENTED** [review]
This PR not only renames a class, it also handles an MSQ-only limitation. Added some suggestions.

File: extensions-core/multi-stage-query/src/test/java/org/apache/druid/msq/test/CalciteSelectQueryMSQTest.java (hunk 1 above)
Comment (adarshsanjeev) [reply extensions-core/multi-stage-query/src/test/java/org/apache/druid/msq/test/CalciteSelectQueryMSQTest.java:174]: Done. Added comments for both tests

**Review by cryptoe — APPROVED** [review]

**Comment by cryptoe** [comment]
@adarshsanjeev Looks like there are some legit test failures.

**Comment by paul-rogers** [comment]
Yes, now that the test actually runs in GHA, we're seeing failures that didn't previously show up.

**Comment by paul-rogers** [comment]
@adarshsanjeev, checking back on this one. There is a merge conflict to resolve and test failures which might need assistance from the folks who wrote the tests.

**Comment by adarshsanjeev** [comment]
Thanks, @paul-rogers! I have resolved the failures and merge conflicts, the PR should be good to go. Regarding the test failures, the cause does not seem to have been introduced by a specific PR (at least not that I can tell), but some discrepancies that have always existed due to how MSQ handles it. I think there are more than these as all tests have not been ported to `CalciteSelectQueryMSQTest` yet.

I have raised two issues to cover these.

https://github.com/apache/druid/issues/13950
https://github.com/apache/druid/issues/13951

I can raise these in the Apache community slack to discuss how to handle these.

**Pull request merged**